api/v1alpha1/azuredisk_types.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// AzureDiskSpec defines the desired state of AzureDisk
+type AzureDiskSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	Location      string `json:"location"`
+	ResourceGroup string `json:"resourceGroup"`
+	CreateOption  string `json:"createOption"`
```

Comment:
Is this field more like an enum? There are ways to describe an enum if you search the codebase for `kubebuilder:validation:Enum` you should find some examples